pkg/controller/scheduler.go

```diff
@@ -939,7 +939,7 @@ func (c *Controller) runMetricChecks(canary *flaggerv1.Canary) bool {
 			}
 
 			factory := providers.Factory{}
-			provider, err := factory.Provider(template.Spec.Provider, credentials)
+			provider, err := factory.Provider(canary, template.Spec.Provider, credentials)
```

Comment:
Here we can set the window size with `factory.Provider(template.Spec.Provider, credentials, metric.interval)`